plugins/liquid/refined-types-plugin/src/main/kotlin/arrow/meta/plugins/liquid/phases/analysis/solver/ConstraintCollection.kt

```diff
@@ -470,15 +483,21 @@ private fun Solver.expressionToFormulae(
   }
 
 internal fun Solver.isResultReference(ex: KtElement, bindingContext: BindingContext): Boolean {
-  val maybePostCall =
+  val maybeParentCall =
     ex.getParentResolvedCall(bindingContext)?.call?.callElement.getParentResolvedCall(bindingContext)
-  return if (maybePostCall != null && maybePostCall.postCall()) {
-    val expArg = maybePostCall.valueArguments.entries.toList()[1].value as? ExpressionValueArgument
+  return when {
+    maybeParentCall?.postCall() == true
+      -> maybeParentCall.valueArguments.entries.toList()[1].value
```

Comment:
I found a couple of cases where we are failing with an array index out of bounds as I was encoding the laws. Regardless of this instance here we need to do a general pass on all these cases and actually check we can access the args in the collections if those elements are there. I presume it was failing because I was referring to fields like `size` and we don't support that yet.

Reply:
I've replaced this code with one that looks for the `predicate` argument, instead of hardcoding the position. In fact, this was already wrong, since we provide different overloads of `post` with different argument order.